tm2/pkg/commands/command.go

```diff
@@ -105,6 +106,19 @@ func (c *Command) AddSubCommands(cmds ...*Command) {
 	}
 }
 
+// Main is a helper function for command entry. It wraps ParseAndRun and
+// handles the flag.ErrHelp error, ensuring that every command with -h or 
+// --help won't show an error message: 
+// 'error parsing commandline arguments: flag: help requested'
+func (c *Command) Main(ctx context.Context, args []string) {
```

Comment:
What do you think about expanding this method to take in a writer (like `os.Stderr`), that handles the command error?

This way we can mock it in tests, and make it inspect-able

Reply:
> What do you think about expanding this method to take in a writer (like `os.Stderr`), that handles the command error?
> 
> This way we can mock it in tests, and make it inspect-able

Hi sir, as there is `os.Exit(1)` inner the "`Main`"(let's call it `Main` temporarily) method, this looks unable to write a test for

Reply:
I agree, I think Execute should be explicitly "what you call from the main function" (we have the whole commands package to mock and test CLI commands anyway :smile:)